lib/Config.js

```diff
@@ -69,21 +69,83 @@ class Config {
         /*
          * By default, the config file is "config.json" at the root.
          * It can be overridden using the S3_CONFIG_FILE environment var.
+         * By default, the locationConfig file is "locationConfig.json" at the
+         * root. It can be overridden using the S3_LOCATION_FILE environment
+         * var.
          */
         this._basePath = path.join(__dirname, '..');
-        this.path = path.join(__dirname, '../config.json');
+        this.configPath = path.join(__dirname, '../config.json');
         if (process.env.S3_CONFIG_FILE !== undefined) {
-            this.path = process.env.S3_CONFIG_FILE;
+            this.configPath = process.env.S3_CONFIG_FILE;
+        }
+        this.locationConfigPath = path.join(__dirname,
+          '../locationConfig.json');
+        if (process.env.S3_LOCATION_FILE !== undefined) {
+            this.locationConfigPath = process.env.S3_LOCATION_FILE;
         }
 
         // Read config automatically
+        this._getLocationConfig();
         this._getConfig();
     }
 
+    _getLocationConfig() {
+        let locationConfig;
+        // check that location config path is valid
+        try {
+            fs.statSync(this.locationConfigPath);
+        } catch (err) {
+            if (err) {
+                // location config path is not valid
+                // legacy config format where only one config file
+                return;
+            }
+        }
+        try {
+            const data = fs.readFileSync(this.locationConfigPath,
+              { encoding: 'utf-8' });
+            locationConfig = JSON.parse(data);
+        } catch (err) {
+            throw new Error(`could not parse location config file:
+              ${err.message}`);
+        }
+
+        this.locationConstraints = {};
```

Comment:
@LaurenSpiegel @bennettbuchanan  Checking if locationConfig is not needed here because 
- if location config path is not valid => return https://github.com/scality/S3/pull/629/files#diff-9280903abc69f09eff9ab36070b6d751R101
- if can not parse it => throw error